### PR TITLE
Add recent release sort option to the website

### PR DESCRIPTION
**Pull request by AhmedBasem20**

### CODE REVIEW

File: aiida-registry-app/src/Components/MainIndex.jsx [hunk 1]
```diff
@@ -55,13 +55,31 @@ function MainIndex() {
...
         sortedPlugins = plugins;
       }
+      else if (option == 'release') {
+        //Sort plugins by the recent release date
+        const pluginsArray = Object.entries(plugins);
+        pluginsArray.sort(([, pluginA], [, pluginB]) => {
+          if (!pluginA.metadata.release_date && !pluginB.metadata.release_date) {
+           return 0; // Both plugins have no release date, keep them in the current order
+          } else if (!pluginA.metadata.release_date) {
+            return 1; // Only pluginB has a release date, so pluginA should come first
```
Comment (unkcpz) [inline aiida-registry-app/src/Components/MainIndex.jsx:72]:
```suggestion
            return 1; // Only pluginB has a release date, put pluginB higher ranking than pluginA.
```

File: aiida-registry-app/src/Components/MainIndex.jsx [hunk 2]
```diff
@@ -55,13 +55,31 @@ function MainIndex() {
...
-      let sortedPlugins;
+      let sortedPlugins = {};
       if (option === 'commits') {
         sortedPlugins = sortByCommits(plugins);
       }
       else if (option == 'alpha') {
         sortedPlugins = plugins;
       }
+      else if (option == 'release') {
+        //Sort plugins by the recent release date
```
Comment (unkcpz) [inline aiida-registry-app/src/Components/MainIndex.jsx:66]:
```suggestion
        // Sort plugins by the recent release date, the newer the release date the larger the value, and the higher ranking it get. Sorting in descending order by date. 
```

File: aiida-registry-app/src/Components/MainIndex.jsx [hunk 3]
```diff
@@ -55,13 +55,31 @@ function MainIndex() {
...
+        const pluginsArray = Object.entries(plugins);
+        pluginsArray.sort(([, pluginA], [, pluginB]) => {
+          if (!pluginA.metadata.release_date && !pluginB.metadata.release_date) {
+           return 0; // Both plugins have no release date, keep them in the current order
+          } else if (!pluginA.metadata.release_date) {
+            return 1; // Only pluginB has a release date, so pluginA should come first
+          } else if (!pluginB.metadata.release_date) {
+            return -1; // Only pluginA has a release date, so pluginB should come first
```
Comment (unkcpz) [inline aiida-registry-app/src/Components/MainIndex.jsx:74]:
```suggestion
            return -1; // Only pluginA has a release date, put pluginA higher ranking than pluginB.
```

**Review by unkcpz — CHANGES_REQUESTED** [review]
Thanks! I resolve the conflict after https://github.com/aiidateam/aiida-registry/pull/255. I think it is all good to go!
Just a nitpick on the comments of the code, I hope it will make the sorting order more clear, do you agree?

**Comment by unkcpz** [comment]
I'll merge `gsoc` to `main` after this one.

**Comment by AhmedBasem20** [comment]
Thanks @unkcpz I corrected the comments you mentioned, and I also moved the code to a separate function `sortByRelease()`. And in `useEffect` I replaced the call to `sortByCommits()` with a call to `handleSort()`.
This way if we want to change the default sort option at any time we just change that line: https://github.com/aiidateam/aiida-registry/blob/82ab0f72205e89e0ca6cd3adfc370ae2eb405ca2/aiida-registry-app/src/Components/MainIndex.jsx#L19

**Review by unkcpz — APPROVED** [review]
Very clear, thanks!

**Pull request merged**